nandtool/nandtool.c

```diff
@@ -177,6 +177,91 @@ static int nandtool_check(void)
 }
 
 
+static int nandtool_dump(const char *outpath, unsigned int start, unsigned int nblocks, int oob)
+{
+	flashsrv_info_t *info;
+	int ret;
+	size_t blocksz, partsz, pagesz;
+	size_t addr, endaddr;
+	uint8_t *buf;
+	int outfd;
+
+	outfd = open(outpath, O_RDWR | O_CREAT);
+	if (outfd < 0) {
+		fprintf(stderr, "nandtool: Fail to open file: %s\n", outpath);
+		return -EINVAL;
+	}
+
+	info = flashmng_info(nandtool_common.oid);
+	if (info == NULL) {
+		close(outfd);
+		fprintf(stderr, "nandtool: Fail to retrieve partition info\n");
+		return -EFAULT;
+	}
+
+	if (!oob) {
+		blocksz = info->erasesz;
```

Comment:
nitpociking: keep the same assignment order in both cases to spot differences easier ;)